.github/workflows/release.yml

```diff
@@ -1,160 +1,251 @@
-      ############## posthog-js auto-update ##################
-      - name: Dispatch posthog upgrade for ${{ matrix.package.name }}
-        if: matrix.package.name == 'posthog-js' && steps.check-package-version.outputs.is-new-version == 'true'
-        run: |
-          curl -f -X POST \
-               -H "Accept: application/vnd.github+json" \
-               -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" \
-               https://api.github.com/repos/posthog/posthog-js/actions/workflows/posthog-upgrade.yml/dispatches \
-               -d '{
-                 "ref": "main",
-                 "inputs": {
-                   "package_name": "${{ matrix.package.name }}",
-                   "package_version": "${{ steps.check-package-version.outputs.committed-version }}"
-                 }
-               }'
-
-      - name: Dispatch posthog.com upgrade for ${{ matrix.package.name }}
-        if: matrix.package.name == 'posthog-js' && steps.check-package-version.outputs.is-new-version == 'true'
-        run: |
-          curl -f -X POST \
-               -H "Accept: application/vnd.github+json" \
-               -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" \
-               https://api.github.com/repos/posthog/posthog-js/actions/workflows/posthog-com-upgrade.yml/dispatches \
-               -d '{
-                 "ref": "main",
-                 "inputs": {
-                   "package_name": "${{ matrix.package.name }}",
-                   "package_version": "${{ steps.check-package-version.outputs.committed-version }}"
-                 }
-               }'
-      ####################################################
-      # https://us.posthog.com/project/11213/functions/019ae9c0-03ee-0000-2f32-971f64be8ffe
-      - name: Send failure event to PostHog
-        if: ${{ failure() }}
-        uses: PostHog/posthog-github-action@v0.1
-        with:
-          posthog-token: "${{ secrets.POSTHOG_PROJECT_API_KEY }}"
-          event: "posthog-js-github-release-workflow-failure"
-          properties: >-
-            {
-              "commitSha": "${{ github.sha }}",
-              "jobStatus": "${{ job.status }}",
-              "commitMessage": "${{ github.event.head_commit.message }}",
-              "commitAuthor": "${{ github.event.head_commit.author.name }}",
-              "ref": "${{ github.ref }}",
-              "packageName": "${{ matrix.package.name }}",
-              "packageVersion": "${{ steps.check-package-version.outputs.committed-version }}"
-            }
+    check-changesets:
+        name: Check for changesets
+        runs-on: ubuntu-latest
+        # Run when PR with 'release' label is merged to main, or when manually triggered
+        if: |
+            github.event_name == 'workflow_dispatch' ||
+            (github.event.pull_request.merged == true
+             && contains(github.event.pull_request.labels.*.name, 'release')
+             && github.event.pull_request.base.ref == 'main')
```

Comment:
I'm pretty sure you can specify all of these requirements directly in `on` above.

Reply:
I based myself off of this: https://docs.github.com/en/actions/reference/workflows-and-actions/events-that-trigger-workflows#running-your-pull_request-workflow-when-a-pull-request-merges  
  
It seems to imply we need to check for merged here

Reply:
The base can be moved over to the global setting :white_check_mark:

Reply:
I thought there was a `merged` type, but you're right there's just closed and then we have to explicitly check if it was merged. And definitely right about labels, oversight on my part.

https://docs.github.com/en/webhooks/webhook-events-and-payloads